keel-ec2-plugin/src/test/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandlerTests.kt

```diff
@@ -287,6 +306,51 @@ internal class ClusterHandlerTests : JUnit5Minutests {
           verify { publisher.publishEvent(ofType<ArtifactVersionDeployed>()) }
         }
       }
+
+      derivedContext<SubmittedResource<ClusterSpec>>("exported cluster spec") {
+        deriveFixture {
+          runBlocking {
+            export(exportable)
+          }
+        }
+
+        test("has the expected basic properties") {
+          expectThat(kind)
+            .isEqualTo("cluster")
+          expectThat(apiVersion)
+            .isEqualTo(SPINNAKER_EC2_API_V1)
+          expectThat(spec.regionalIds)
+            .hasSize(2)
+          expectThat(spec.locations.regions)
+            .hasSize(2)
+          expectThat(spec.overrides)
+            .hasSize(0)
+        }
+
+        test("omits complex fields altogether when all their properties have default values") {
+          expectThat(spec.defaults.health)
+            .isNull()
+          expectThat(spec.defaults.scaling)
+            .isNull()
+        }
+      }
+
+      context("other handling of default properties in cluster export") {
+        before {
+          coEvery { cloudDriverService.activeServerGroup("us-east-1") } returns activeServerGroupResponseEast
+          coEvery { cloudDriverService.activeServerGroup("us-west-2") } returns activeServerGroupResponseWest.withNonDefaultHealthProps()
+        }
+
+        test("export omits properties with default values from complex fields") {
```

Comment:
@robfletcher I tried to add this test under the `derivedContext` up above, but was unable to override the mocked return for `cloudDriverService.activeServerGroup("us-west-2")` that you see here. Please let me know if there's a better way to do this.